### PR TITLE
User list search, status and role

### PR DESCRIPTION
**Pull request by AlixH**
- [x] Fixed lint issues
- [x] Added user status and role
- [x] Added search input

### CODE REVIEW

File: src/screens/users/list/UsersList.tsx [hunk 1]
```diff
@@ -123,6 +125,10 @@ export default class UsersList extends BaseAutoRefreshScreen<Props, State> {
       });
     }
   }
+  public search = async (searchText: string) => {
```
Comment (LucasBrazi06) [inline src/screens/users/list/UsersList.tsx:128]: Missing blank line :-)

File: src/types/User.tsx [hunk 2]
```diff
@@ -51,3 +51,18 @@ export interface UserSite {
   siteID: string;
   siteAdmin: boolean;
 }
+
+export enum UserStatus {
```
Comment (LucasBrazi06) [inline src/types/User.tsx:55]: Use the same type as in the backend

File: src/types/User.tsx [hunk 3]
```diff
@@ -51,3 +51,18 @@ export interface UserSite {
...
+  P = 'PENDING',
+  A = 'ACTIVE',
+  I = 'INACTIVE',
+  B = 'BLOCKED',
+  L = 'LOCKED'
+}
+
+export enum UserRole {
```
Comment (LucasBrazi06) [inline src/types/User.tsx:63]: Use the same type as in the backend

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/components/user/UserComponent.tsx [hunk 4]
```diff
@@ -26,11 +28,25 @@ export default class UserComponent extends React.Component<Props, State>{
...
+        default: return style.inactive;
+      }
+    }
+    return null;
+  }
+
   public render() {
     const style = computeStyleSheet();
     const { user, selected, navigation } = this.props;
     const userName = user.name ? (user.name.charAt(0).toUpperCase() + user.name.slice(1).toLowerCase()) : '';
```
[the remark this answers is not in the document]
Comment (AlixH) [reply src/components/user/UserComponent.tsx:45]: You mean in the UserAvatar component ?

File: src/components/user/UserComponent.tsx [hunk 5]
```diff
@@ -26,11 +28,25 @@ export default class UserComponent extends React.Component<Props, State>{
     super.setState(state, callback);
   }
 
+  private computeStatusStyle(status: UserStatus, style: any) {
+    if (status) {
+      switch (status) {
+        case UserStatus.A: return style.active; break;
```
[the remark this answers is not in the document]
Comment (AlixH) [reply src/components/user/UserComponent.tsx:34]: The backend returns only A, P, L, B, I, not the full words, I did not managed to map the letters to the enum word values

File: src/components/user/UserComponent.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/components/user/UserComponent.tsx:45]: Yes the two letters in the circle

File: src/components/user/UserComponent.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/components/user/UserComponent.tsx:34]: I'm talking about the enum name:

```
export enum UserStatus {
  PENDING = 'P',
  ACTIVE = 'A',
  INACTIVE = 'I',
  BLOCKED = 'B',
  LOCKED = 'L',
}
```

[the remark this answers is not in the document]
Comment (AlixH) [reply src/components/user/UserComponent.tsx:34]: So I do the reverse mapping by hand ? 
My pb is how to tranform 'P' to 'PENDING' for example

[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/components/user/UserComponent.tsx:34]: > My pb is how to tranform 'P' to 'PENDING' for example

Why is it a problem?

**Review by LucasBrazi06 — APPROVED** [review]
GUT!

**Pull request merged**